lib/msal-core/src/BrowserCrypto.ts

```diff
@@ -0,0 +1,134 @@
+import { KeyGenAlgorithm, HashAlgorithm, KeyFormat, getArrayBufferFromString, getStringFromArrayBuffer, isIE11, getJwkString } from './PopTokenCommon';
+
+/**
+ * Thin wrapper around browser Web Crypto APIs.
+ * Docs: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto
+ *
+ * Includes support for IE11 msCrypto methods.
+ * Docs: https://msdn.microsoft.com/en-us/ie/dn904640(v=vs.94)
+ */
+export class BrowserCrypto {
+    private _algorithmOptions: RsaHashedKeyGenParams;
+
+    constructor(keyGenAlgorithm: KeyGenAlgorithm, hashAlgorithm: HashAlgorithm, modulusLength: number, publicExponent: Uint8Array) {
+        this._algorithmOptions = {
+            name: keyGenAlgorithm,
+            hash: hashAlgorithm,
+            modulusLength,
+            publicExponent
+        }
+    }
+
+    async sign(key: CryptoKey, data: ArrayBuffer): Promise<ArrayBuffer> {
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const signOperation = window['msCrypto'].subtle.sign(this._algorithmOptions, key, data);
+                signOperation.addEventListener('complete', (e: { target: { result: ArrayBuffer | PromiseLike<ArrayBuffer>; }; }) => {
+                    resolve(e.target.result);
+                });
+                signOperation.addEventListener('error', (error: any) => {
+                    reject(error);
+                });
+            });
+        }
+
+        return window.crypto.subtle.sign(this._algorithmOptions, key, data);
+    }
+
+    async verify(key: CryptoKey, signature: ArrayBuffer, data: ArrayBuffer): Promise<boolean> {
+        if (isIE11()) {
+            return new Promise((resolve, reject) => {
+                const verifySignatureOperation = window['msCrypto'].subtle.verify(this._algorithmOptions, key, signature, data);
+                verifySignatureOperation.addEventListener('complete', (e: { target: { result: boolean | PromiseLike<boolean>; }; }) => {
+                    resolve(e.target.result);
+                });
+                verifySignatureOperation.addEventListener('error', (error: any) => {
+                    reject(error);
+                });
+            });
+        }
+
+        return window.crypto.subtle.verify(this._algorithmOptions, key, signature, data);
+    }
+
+    async generateKey(extractable: boolean, usages: Array<string>): Promise<CryptoKeyPair> {
+        if (isIE11()) {
```

Comment:
maybe breakout the if(isIe11()) then do this logic in to functions so that your top level apis read cleaner 

```js
async generateKey(extractable: boolean, usages: Array<string>): Promise<CryptoKeyPair> {
  return isIE11() ? generateIe11Key(extractable. usages) : window.crypto.subtle.generateKey(this._algorithmOptions, extractable, usages);

}
```

Reply:
What about sub IE 11?

Reply:
Yeah, I could probably create an abstraction for the `msCrypto` operations, might revisit that. 

And if a browser doesn't have either `window.crypto` or `window.msCrypto` (which is in IE11 only, I believe), it will not be able to support PoP.

There is also `webkitSubtleCrypto` [in older Safari versions](https://webkit.org/blog/7790/update-on-web-cryptography/), but I don't think I'm going to worry about supporting that.